packages/block-editor/src/components/block-removal-warning-modal/index.js

```diff
@@ -41,6 +41,19 @@ export function BlockRemovalWarningModal( { rules } ) {
 		return;
 	}
 
+	const message =
+		messageType === 'templates'
+			? _n(
+					'Deleting this block will stop your post or page content from displaying on this template. It is not recommended.',
+					'Deleting these blocks will stop your post or page content from displaying on this template. It is not recommended.',
+					blockNamesForPrompt.length
+			  )
+			: _n(
+					'Deleting this block could break patterns on your site that have content linked to it. Are you sure you want to delete it?',
+					'Deleting these blocks could break patterns on your site that have content linked to them. Are you sure you want to delete them?',
+					blockNamesForPrompt.length
+			  );
```

Comment:
I think it could mention something like "This block allows instance overrides, deleting it ...", just so that the language used in the interface is consistent.